File: VB/E3422/IDialogService.vb
Imports System
Imports System.IO

Namespace E3422
    Public Interface IDialogService
        Sub ShowPreview(ByVal serviceUri As String, ByVal reportName As String)
        Function ShowSaveFileDialog(ByVal filter As String) As Stream
        Sub ShowMessage(ByVal caption As String, ByVal message As String)
        Sub OpenBrowserWindow(ByVal uri As Uri)
        Sub AsyncRequestPrintingConfirmation(ByVal continuePrinting As Action(Of Boolean))
    End Interface
End Namespace

File: CS/E3422/IDialogService.cs
using System;
using System.IO;

namespace E3422 {
    public interface IDialogService {
        void ShowPreview(string serviceUri, string reportName);
        Stream ShowSaveFileDialog(string filter);
        void ShowMessage(string caption, string message);
        void OpenBrowserWindow(Uri uri);
        void AsyncRequestPrintingConfirmation(Action<bool> continuePrinting);
    }
}
